src/utility.cpp

```diff
@@ -423,38 +423,81 @@ BOOL IsModulePatched (HMODULE importmodule, moduleentry_t patchtable [], UINT ta
 
     // No patches listed in the patch table were found in the import module.
     return FALSE;
-}
-
+}
+
 LPVOID FindRealCode(LPVOID pCode)
-{
-    LPVOID result = pCode;
+{
+    LPVOID result;
     if (pCode != NULL)
     {
-        if (*(WORD *)pCode == 0x25ff) // JMP r/m32
-        {
+        // we need to make sure we can read the first 3 ULONG_PTRs
+        DWORD old_protect;
+        // make sure we can read the first 3 pointers
```

Comment:
>make sure we can read the first 3 pointers [](start = 11, length = 42)

nit: same as last comment

Reply:
Removed

---
In reply to: [591983102](https://github.com/Azure/vld/pull/8#discussion_r591983102) [](ancestors = 591983102)